analyzer/runtime/extract.go

```diff
@@ -364,15 +364,42 @@ func ExtractRound(blockHeader nodeapi.RuntimeBlockHeader, txrs []nodeapi.Runtime
 					blockTransactionData.Body = body
 					amount = body.Amount.Amount
 					if body.To != nil {
-						// Beware, this is the address of an account in the consensus
-						// layer, not an account in the runtime that generated this event.
+						// This is the address of an account in the consensus
+						// layer, not an account in the runtime that generated this tx.
 						// We do not register it as a preimage.
 						if to, err = uncategorized.StringifySdkAddress(body.To); err != nil {
 							return fmt.Errorf("to: %w", err)
 						}
 					}
 					return nil
 				},
+				ConsensusAccountsDelegate: func(body *consensusaccounts.Delegate) error {
```

Comment:
This should probably check if the event contains an error https://github.com/oasisprotocol/oasis-sdk/blob/fac9e70f9444e0f20e62ca56cc597abfd3f54e38/client-sdk/go/modules/consensusaccounts/types.go#L201


Actually all `consensus_account` events can contain a potential consensus error in which indicates a failed event, and should probably be skipped here?

See: https://github.com/oasisprotocol/oasis-sdk/blob/fac9e70f9444e0f20e62ca56cc597abfd3f54e38/client-sdk/go/modules/consensusaccounts/types.go#L233-L239

and: https://github.com/oasisprotocol/oasis-sdk/blob/fac9e70f9444e0f20e62ca56cc597abfd3f54e38/runtime-sdk/src/modules/consensus_accounts/mod.rs#L609-L625

for examples.

Reply:
Hmm ... it does feel weird to basically ignore errors, but I'm not sure the indexer needs to do anything special about `consensusaccounts` events with a non-nil `Error`:
- There is no dead reckoning to adjust (or suppress); that all happens via other, co-emitted events
- We [store](https://github.com/oasisprotocol/nexus/blob/f9aae40e235f3dca08bb2845f8fb124603b8f310/analyzer/runtime/extract.go#L397-L397) (and then expose via API) the entire body of the event as JSON; [example](https://index-staging.oasislabs.com/v1/emerald/events?type=consensus_accounts.deposit). If the `Error` field were set, we'd include it there.

Maybe I'm forgetting something?

Reply:
You're right